etcdserver/cindex/cindex.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2015 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cindex
+
+import (
+	"encoding/binary"
+	"sync"
+	"sync/atomic"
+
+	"go.etcd.io/etcd/mvcc/backend"
+)
+
+var (
+	metaBucketName = []byte("meta")
+
+	consistentIndexKeyName = []byte("consistent_index")
+)
+
+// ConsistentIndexer is an interface that wraps the Get/Save method.
+// Consistent index is the offset of an entry in a consistent replicated log.
+type ConsistentIndexer interface {
+
+	// ConsistentIndex returns the consistent index of current executing entry.
+	ConsistentIndex() uint64
+
+	// SetConsistentIndex set the consistent index of current executing entry.
+	SetConsistentIndex(v uint64)
+
+	// UnsafeSave saves consistentIndex to the underlying stable storage.
```

Comment:
Can we add additional comment explaining this function must be called holding the lock on the tx?

Reply:
updated. thanks.